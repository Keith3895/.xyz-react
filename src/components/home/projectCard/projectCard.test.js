import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import renderer from 'react-test-renderer';
import { act } from "react-dom/test-utils";
import pretty from "pretty";
import Project from './projectCard';
import { shallow } from 'enzyme';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { APIresponse } from './__snapshots__/resp';
let container = null;
let spy;
// jest.mock('axios');
beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;

});
// it('snap shot test', () => {
//     act(() => {
//         render(
//             <Project />
//             , container);
//     });
//     expect(pretty(container.innerHTML)).toMatchSnapshot();
// });

describe('getInitails', () => {
    beforeEach(() => {
        var mock = new MockAdapter(axios);
        const data = { response: true };
        mock.onGet('https://api.github.com/users/keith3895/starred?per_page=6').reply(200, data);
    });
    it('positive test of _getUserDetails with "Keith Franklin"', () => {
        let wrapper = shallow(<Project />);
        const instance = wrapper.instance();
        let resp = instance._getInitials('Keith Franklin');
        expect(resp).toBe('KF');
    });
    it('positive test of _getUserDetails with "Keith-Franklin"', () => {
        let wrapper = shallow(<Project />);
        const instance = wrapper.instance();
        let resp = instance._getInitials('Keith-Franklin');
        expect(resp).toBe('KF');
    });
    it('positive test of _getUserDetails with "Keith_Franklin"', () => {
        let wrapper = shallow(<Project />);
        const instance = wrapper.instance();
        let resp = instance._getInitials('Keith-Franklin');
        expect(resp).toBe('KF');
    });
    it('positive test of _getUserDetails with "OneWord"', () => {
        let wrapper = shallow(<Project />);
        const instance = wrapper.instance();
        let resp = instance._getInitials('OneWord');
        expect(resp).toBe('ON');
    });
    it('positive test of _getUserDetails with ""', () => {
        let wrapper = shallow(<Project />);
        const instance = wrapper.instance();
        let resp = instance._getInitials('');
        expect(resp).toBe('');
    });
    it('positive test of _getUserDetails with null|undefined', () => {
        let wrapper = shallow(<Project />);
        const instance = wrapper.instance();
        let resp = instance._getInitials(null);
        expect(resp).toBe('');
        resp = instance._getInitials(undefined);
        expect(resp).toBe('');
    });
    it('negative test of _getUserDetails with "Super Admin" ', () => {
        let wrapper = shallow(<Project />);
        const instance = wrapper.instance();
        let resp = instance._getInitials('Super Admin');
        expect(resp).not.toBe('KF');
    });
});

describe('getProjects', () => {
    var mock;
    beforeEach(() => {
        mock = new MockAdapter(axios);
        mock.onGet('https://api.github.com/users/keith3895/starred?per_page=6').reply(200, APIresponse);
    });
    it('axios mock and check if call is made', () => {
        // spy.then((res)=>{
        //     console.log(res);
        // });
        let wrapper = shallow(<Project />);
        const instance = wrapper.instance();
        instance._getProject().then(res => {
            expect(mock.history.get.length).toBe(2);
            expect(instance.state['projects']).toStrictEqual(APIresponse);
        }).catch(e => expect(e).toMatch('error'));
    });

});

describe('languageColor', () => {
    it('with js', () => {
        let wrapper = shallow(<Project />);
        const instance = wrapper.instance();
        let resp = instance.languageColor('JavaScript');
        const tree = renderer
            .create(resp)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('with TS', () => {
        let wrapper = shallow(<Project />);
        const instance = wrapper.instance();
        let resp = instance.languageColor('TypeScript');
        const tree = renderer
            .create(resp)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('with ""', () => {
        let wrapper = shallow(<Project />);
        const instance = wrapper.instance();
        let resp = instance.languageColor('');
        const tree = renderer
            .create(resp)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('showCard', () => {
    it('with empty list', () => {
        let wrapper = shallow(<Project />);
        const instance = wrapper.instance();
        let resp = instance.showCard([]);
        const tree = renderer
            .create(resp)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('with full list', () => {
        let wrapper = shallow(<Project />);
        const instance = wrapper.instance();
        let resp = instance.showCard(APIresponse);
        const tree = renderer
            .create(resp)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});