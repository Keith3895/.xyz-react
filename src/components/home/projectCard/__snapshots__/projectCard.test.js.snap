// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`languageColor with "" 1`] = `""`;

exports[`languageColor with TS 1`] = `
<div
  style={
    Object {
      "width": "50%",
    }
  }
>
  <svg
    height="10"
    width="10"
  >
    <circle
      cx="50%"
      cy="50%"
      fill="#2b7489"
      r="5"
    />
  </svg>
  <span>
     
    TypeScript
  </span>
</div>
`;

exports[`languageColor with js 1`] = `
<div
  style={
    Object {
      "width": "50%",
    }
  }
>
  <svg
    height="10"
    width="10"
  >
    <circle
      cx="50%"
      cy="50%"
      fill="#f1e05a"
      r="5"
    />
  </svg>
  <span>
     
    JavaScript
  </span>
</div>
`;

exports[`showCard with empty list 1`] = `""`;

exports[`showCard with full list 1`] = `
Array [
  <div
    className="cardContainer"
  >
    <div
      className="card"
    >
      <div
        className="flexCenter cardImageContainer"
      >
        <svg
          className="cardImage"
          height="100"
          width="100"
          xlink="http://www.w3.org/1999/xlink"
        >
          <circle
            cx="50%"
            cy="50%"
            fill="white"
            r="50"
          />
          <text
            alignmentBaseline="middle"
            fill="#000"
            fontSize="50"
            textAnchor="middle"
            x="50%"
            y="50%"
          >
            PU
          </text>
        </svg>
      </div>
      <a
        href="https://github.com/Keith3895/pubSub"
        rel="noopener noreferrer"
        style={
          Object {
            "color": "unset",
          }
        }
        target="_blank"
      >
        <div
          className="card-content"
        >
          <h3>
            pubSub
          </h3>
          <p
            style={
              Object {
                "padding": "15px",
              }
            }
          >
            The Goal of this module is to create a simple mechanism to achieve Pub-Sub messaging services. The nodes can be used to communicate between flows also.
          </p>
          <div
            className="flexColumn"
            style={
              Object {
                "color": "#586069",
              }
            }
          >
            <div>
              Last Update on 
              Thu, 18 Jun 2020 17:03:02 GMT
            </div>
            
            <div
              className="flexRow"
              style={
                Object {
                  "alignContent": "space-between",
                  "alignItems": "center",
                  "flexWrap": "wrap",
                  "padding": "1em",
                }
              }
            >
              <div
                style={
                  Object {
                    "width": "50%",
                  }
                }
              >
                <svg
                  height="10"
                  width="10"
                >
                  <circle
                    cx="50%"
                    cy="50%"
                    fill="#f1e05a"
                    r="5"
                  />
                </svg>
                <span>
                   
                  JavaScript
                </span>
              </div>
              <div
                style={
                  Object {
                    "width": "25%",
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-star fa-w-18 "
                  data-icon="star"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={
                    Object {
                      "marginRight": "5px",
                    }
                  }
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                1
              </div>
              <div
                style={
                  Object {
                    "width": "25%",
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-eye fa-w-18 "
                  data-icon="eye"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={
                    Object {
                      "marginRight": "5px",
                    }
                  }
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                1
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>,
  <div
    className="cardContainer"
  >
    <div
      className="card"
    >
      <div
        className="flexCenter cardImageContainer"
      >
        <svg
          className="cardImage"
          height="100"
          width="100"
          xlink="http://www.w3.org/1999/xlink"
        >
          <circle
            cx="50%"
            cy="50%"
            fill="white"
            r="50"
          />
          <text
            alignmentBaseline="middle"
            fill="#000"
            fontSize="50"
            textAnchor="middle"
            x="50%"
            y="50%"
          >
            FI
          </text>
        </svg>
      </div>
      <a
        href="https://github.com/Keith3895/financeTracker"
        rel="noopener noreferrer"
        style={
          Object {
            "color": "unset",
          }
        }
        target="_blank"
      >
        <div
          className="card-content"
        >
          <h3>
            financeTracker
          </h3>
          <p
            style={
              Object {
                "padding": "15px",
              }
            }
          >
            A web application to track the finance of the user.
          </p>
          <div
            className="flexColumn"
            style={
              Object {
                "color": "#586069",
              }
            }
          >
            <div>
              Last Update on 
              Thu, 18 Jun 2020 14:27:16 GMT
            </div>
            <div
              style={
                Object {
                  "alignContent": "space-around",
                  "alignItems": "center",
                  "alignSelf": "center",
                  "display": "flex",
                  "flexWrap": "wrap",
                  "justifyContent": "center",
                  "padding": "1em",
                }
              }
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-balance-scale fa-w-20 "
                data-icon="balance-scale"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={
                  Object {
                    "color": "#586069",
                    "width": "10%",
                  }
                }
                viewBox="0 0 640 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 336h-.02c0-16.18 1.34-8.73-85.05-181.51-17.65-35.29-68.19-35.36-85.87 0C-2.06 328.75.02 320.33.02 336H0c0 44.18 57.31 80 128 80s128-35.82 128-80zM128 176l72 144H56l72-144zm511.98 160c0-16.18 1.34-8.73-85.05-181.51-17.65-35.29-68.19-35.36-85.87 0-87.12 174.26-85.04 165.84-85.04 181.51H384c0 44.18 57.31 80 128 80s128-35.82 128-80h-.02zM440 320l72-144 72 144H440zm88 128H352V153.25c23.51-10.29 41.16-31.48 46.39-57.25H528c8.84 0 16-7.16 16-16V48c0-8.84-7.16-16-16-16H383.64C369.04 12.68 346.09 0 320 0s-49.04 12.68-63.64 32H112c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h129.61c5.23 25.76 22.87 46.96 46.39 57.25V448H112c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h416c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
              Apache License 2.0
            </div>
            <div
              className="flexRow"
              style={
                Object {
                  "alignContent": "space-between",
                  "alignItems": "center",
                  "flexWrap": "wrap",
                  "padding": "1em",
                }
              }
            >
              <div
                style={
                  Object {
                    "width": "50%",
                  }
                }
              >
                <svg
                  height="10"
                  width="10"
                >
                  <circle
                    cx="50%"
                    cy="50%"
                    fill="#2b7489"
                    r="5"
                  />
                </svg>
                <span>
                   
                  TypeScript
                </span>
              </div>
              <div
                style={
                  Object {
                    "width": "25%",
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-star fa-w-18 "
                  data-icon="star"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={
                    Object {
                      "marginRight": "5px",
                    }
                  }
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                3
              </div>
              <div
                style={
                  Object {
                    "width": "25%",
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-eye fa-w-18 "
                  data-icon="eye"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={
                    Object {
                      "marginRight": "5px",
                    }
                  }
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                3
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>,
  <div
    className="cardContainer"
  >
    <div
      className="card"
    >
      <div
        className="flexCenter cardImageContainer"
      >
        <svg
          className="cardImage"
          height="100"
          width="100"
          xlink="http://www.w3.org/1999/xlink"
        >
          <circle
            cx="50%"
            cy="50%"
            fill="white"
            r="50"
          />
          <text
            alignmentBaseline="middle"
            fill="#000"
            fontSize="50"
            textAnchor="middle"
            x="50%"
            y="50%"
          >
            ST
          </text>
        </svg>
      </div>
      <a
        href="https://github.com/Keith3895/straightedge"
        rel="noopener noreferrer"
        style={
          Object {
            "color": "unset",
          }
        }
        target="_blank"
      >
        <div
          className="card-content"
        >
          <h3>
            straightedge
          </h3>
          <p
            style={
              Object {
                "padding": "15px",
              }
            }
          >
            Straightedge is(will be) a application that will generate code that can be run by mocha to perform end to end tests.
          </p>
          <div
            className="flexColumn"
            style={
              Object {
                "color": "#586069",
              }
            }
          >
            <div>
              Last Update on 
              Thu, 18 Jun 2020 17:01:58 GMT
            </div>
            <div
              style={
                Object {
                  "alignContent": "space-around",
                  "alignItems": "center",
                  "alignSelf": "center",
                  "display": "flex",
                  "flexWrap": "wrap",
                  "justifyContent": "center",
                  "padding": "1em",
                }
              }
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-balance-scale fa-w-20 "
                data-icon="balance-scale"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={
                  Object {
                    "color": "#586069",
                    "width": "10%",
                  }
                }
                viewBox="0 0 640 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 336h-.02c0-16.18 1.34-8.73-85.05-181.51-17.65-35.29-68.19-35.36-85.87 0C-2.06 328.75.02 320.33.02 336H0c0 44.18 57.31 80 128 80s128-35.82 128-80zM128 176l72 144H56l72-144zm511.98 160c0-16.18 1.34-8.73-85.05-181.51-17.65-35.29-68.19-35.36-85.87 0-87.12 174.26-85.04 165.84-85.04 181.51H384c0 44.18 57.31 80 128 80s128-35.82 128-80h-.02zM440 320l72-144 72 144H440zm88 128H352V153.25c23.51-10.29 41.16-31.48 46.39-57.25H528c8.84 0 16-7.16 16-16V48c0-8.84-7.16-16-16-16H383.64C369.04 12.68 346.09 0 320 0s-49.04 12.68-63.64 32H112c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h129.61c5.23 25.76 22.87 46.96 46.39 57.25V448H112c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h416c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
              GNU General Public License v3.0
            </div>
            <div
              className="flexRow"
              style={
                Object {
                  "alignContent": "space-between",
                  "alignItems": "center",
                  "flexWrap": "wrap",
                  "padding": "1em",
                }
              }
            >
              <div
                style={
                  Object {
                    "width": "50%",
                  }
                }
              >
                <svg
                  height="10"
                  width="10"
                >
                  <circle
                    cx="50%"
                    cy="50%"
                    fill="#f1e05a"
                    r="5"
                  />
                </svg>
                <span>
                   
                  JavaScript
                </span>
              </div>
              <div
                style={
                  Object {
                    "width": "25%",
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-star fa-w-18 "
                  data-icon="star"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={
                    Object {
                      "marginRight": "5px",
                    }
                  }
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                1
              </div>
              <div
                style={
                  Object {
                    "width": "25%",
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-eye fa-w-18 "
                  data-icon="eye"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={
                    Object {
                      "marginRight": "5px",
                    }
                  }
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                1
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>,
  <div
    className="cardContainer"
  >
    <div
      className="card"
    >
      <div
        className="flexCenter cardImageContainer"
      >
        <svg
          className="cardImage"
          height="100"
          width="100"
          xlink="http://www.w3.org/1999/xlink"
        >
          <circle
            cx="50%"
            cy="50%"
            fill="white"
            r="50"
          />
          <text
            alignmentBaseline="middle"
            fill="#000"
            fontSize="50"
            textAnchor="middle"
            x="50%"
            y="50%"
          >
            NJ
          </text>
        </svg>
      </div>
      <a
        href="https://github.com/Keith3895/node-red-jbpm"
        rel="noopener noreferrer"
        style={
          Object {
            "color": "unset",
          }
        }
        target="_blank"
      >
        <div
          className="card-content"
        >
          <h3>
            node-red-jbpm
          </h3>
          <p
            style={
              Object {
                "padding": "15px",
              }
            }
          >
            The Goal of this module is to create a simple and modular approach to making Remote API Calls to jBPM Workbench.
          </p>
          <div
            className="flexColumn"
            style={
              Object {
                "color": "#586069",
              }
            }
          >
            <div>
              Last Update on 
              Thu, 18 Jun 2020 17:11:23 GMT
            </div>
            
            <div
              className="flexRow"
              style={
                Object {
                  "alignContent": "space-between",
                  "alignItems": "center",
                  "flexWrap": "wrap",
                  "padding": "1em",
                }
              }
            >
              <div
                style={
                  Object {
                    "width": "50%",
                  }
                }
              >
                <svg
                  height="10"
                  width="10"
                >
                  <circle
                    cx="50%"
                    cy="50%"
                    fill="#f1e05a"
                    r="5"
                  />
                </svg>
                <span>
                   
                  JavaScript
                </span>
              </div>
              <div
                style={
                  Object {
                    "width": "25%",
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-star fa-w-18 "
                  data-icon="star"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={
                    Object {
                      "marginRight": "5px",
                    }
                  }
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                3
              </div>
              <div
                style={
                  Object {
                    "width": "25%",
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-eye fa-w-18 "
                  data-icon="eye"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={
                    Object {
                      "marginRight": "5px",
                    }
                  }
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                3
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>,
  <div
    className="cardContainer"
  >
    <div
      className="card"
    >
      <div
        className="flexCenter cardImageContainer"
      >
        <svg
          className="cardImage"
          height="100"
          width="100"
          xlink="http://www.w3.org/1999/xlink"
        >
          <circle
            cx="50%"
            cy="50%"
            fill="white"
            r="50"
          />
          <text
            alignmentBaseline="middle"
            fill="#000"
            fontSize="50"
            textAnchor="middle"
            x="50%"
            y="50%"
          >
            LI
          </text>
        </svg>
      </div>
      <a
        href="https://github.com/Keith3895/LicensePlateRecognition"
        rel="noopener noreferrer"
        style={
          Object {
            "color": "unset",
          }
        }
        target="_blank"
      >
        <div
          className="card-content"
        >
          <h3>
            LicensePlateRecognition
          </h3>
          <p
            style={
              Object {
                "padding": "15px",
              }
            }
          >
            
          </p>
          <div
            className="flexColumn"
            style={
              Object {
                "color": "#586069",
              }
            }
          >
            <div>
              Last Update on 
              Thu, 18 Jun 2020 14:06:01 GMT
            </div>
            
            <div
              className="flexRow"
              style={
                Object {
                  "alignContent": "space-between",
                  "alignItems": "center",
                  "flexWrap": "wrap",
                  "padding": "1em",
                }
              }
            >
              <div
                style={
                  Object {
                    "width": "50%",
                  }
                }
              >
                <svg
                  height="10"
                  width="10"
                >
                  <circle
                    cx="50%"
                    cy="50%"
                    fill="#3572A5"
                    r="5"
                  />
                </svg>
                <span>
                   
                  Python
                </span>
              </div>
              <div
                style={
                  Object {
                    "width": "25%",
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-star fa-w-18 "
                  data-icon="star"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={
                    Object {
                      "marginRight": "5px",
                    }
                  }
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                1
              </div>
              <div
                style={
                  Object {
                    "width": "25%",
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-eye fa-w-18 "
                  data-icon="eye"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={
                    Object {
                      "marginRight": "5px",
                    }
                  }
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                1
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>,
  <div
    className="cardContainer"
  >
    <div
      className="card"
    >
      <div
        className="flexCenter cardImageContainer"
      >
        <svg
          className="cardImage"
          height="100"
          width="100"
          xlink="http://www.w3.org/1999/xlink"
        >
          <circle
            cx="50%"
            cy="50%"
            fill="white"
            r="50"
          />
          <text
            alignmentBaseline="middle"
            fill="#000"
            fontSize="50"
            textAnchor="middle"
            x="50%"
            y="50%"
          >
            NR
          </text>
        </svg>
      </div>
      <a
        href="https://github.com/Keith3895/node-red-contrib-ntlm-http-request"
        rel="noopener noreferrer"
        style={
          Object {
            "color": "unset",
          }
        }
        target="_blank"
      >
        <div
          className="card-content"
        >
          <h3>
            node-red-contrib-ntlm-http-request
          </h3>
          <p
            style={
              Object {
                "padding": "15px",
              }
            }
          >
            A Node-red module to trigger the HTTP Request with the help of NTLM authentication.
          </p>
          <div
            className="flexColumn"
            style={
              Object {
                "color": "#586069",
              }
            }
          >
            <div>
              Last Update on 
              Thu, 18 Jun 2020 17:04:30 GMT
            </div>
            
            <div
              className="flexRow"
              style={
                Object {
                  "alignContent": "space-between",
                  "alignItems": "center",
                  "flexWrap": "wrap",
                  "padding": "1em",
                }
              }
            >
              <div
                style={
                  Object {
                    "width": "50%",
                  }
                }
              >
                <svg
                  height="10"
                  width="10"
                >
                  <circle
                    cx="50%"
                    cy="50%"
                    fill="#f1e05a"
                    r="5"
                  />
                </svg>
                <span>
                   
                  JavaScript
                </span>
              </div>
              <div
                style={
                  Object {
                    "width": "25%",
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-star fa-w-18 "
                  data-icon="star"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={
                    Object {
                      "marginRight": "5px",
                    }
                  }
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                1
              </div>
              <div
                style={
                  Object {
                    "width": "25%",
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-eye fa-w-18 "
                  data-icon="eye"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={
                    Object {
                      "marginRight": "5px",
                    }
                  }
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                1
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>,
]
`;
